src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
More importantly, as I showed earlier, writing to the temp resource takes a fair amount of code (more than 1 line) and is (imo) one of the less obvious bits of functionality in PHP.

However, I would again defer to the idea that this package ship with a `create_temporary_resouce($string)` function, so that implementations don't have to reinvent the wheel, which leads us back to:

```php
createStream($string_or_resource)
```

I keep coming back to this point because streams are one of the few things that will almost certainly be used in at the application level and not at the package level. As such, it seems reasonable to allow for this kind of usage:

```php
$content = some_domain_action($request_input);

$response = $this->responseFactory->createResponse(200);
$body = $this->streamFactory->createStream($content);

return $response->withBody($body);
```

The alternative requires extra code that I don't feel it is necessary to impose on users:

```php
$content = some_domain_action($request_input);

$response = $this->responseFactory->createResponse(200);

if (is_string($content)) {
    $body = $this->streamFactory->createStreamFromString($content);
} else {
    $body = $this->streamFactory->createStreamFromResource($content);
}

return $response->withBody($body);
```

If there were separate factories for each of these, it would be no different. The only way to reduce the amount of user-land code is to allow for `string|resource`.